src/parsers/manifestjson.js

```diff
@@ -719,6 +719,15 @@ export default class ManifestJSONParser extends JSONParser {
       this.validateHomePageURL(this.parsedJSON.homepage_url);
     }
 
+    if (
+      this.selfHosted &&
+      this.parsedJSON.manifest_version > 2 &&
+      !this.parsedJSON.install_origins
```

Comment:
Where did we land on authorizing an empty array for self-distributed without a host at all ? IIRC Firefox was supposed to let you do `install_origins: []` and we'd let that through, maybe with a warning, for people who want to just drag & drop the xpi locally and such.

Reply:
Eh, past me is smarter than current me